api/pkg/imagebuilder/imagebuilder_test.go

```diff
@@ -1867,3 +1868,190 @@ func Test_imageBuilder_getHashedModelDependenciesUrl(t *testing.T) {
+				job := &batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%s-%s", project.Name, model.Name, modelVersion.ID),
+						Namespace: config.BuildNamespace,
+					},
+					Status: batchv1.JobStatus{
+						Succeeded: 1,
+					},
+				}
+				return true, job, nil
+			},
+			want: models.ImageBuildingJobStatus{
+				State: models.ImageBuildingJobStateSucceeded,
+			},
+			wantErr: false,
+		},
+		{
+			name: "active",
+			args: args{
+				project: project,
+				model:   model,
+				version: modelVersion,
+			},
+			mockGetJob: func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				job := &batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%s-%s", project.Name, model.Name, modelVersion.ID),
+						Namespace: config.BuildNamespace,
+					},
+					Status: batchv1.JobStatus{
+						Active: 1,
+					},
+				}
+				return true, job, nil
+			},
+			want: models.ImageBuildingJobStatus{
+				State: models.ImageBuildingJobStateActive,
+			},
+			wantErr: false,
+		},
+		{
+			name: "failed",
+			args: args{
+				project: project,
+				model:   model,
+				version: modelVersion,
+			},
+			mockGetJob: func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				job := &batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%s-%s", project.Name, model.Name, modelVersion.ID),
+						Namespace: config.BuildNamespace,
+					},
+					Status: batchv1.JobStatus{
+						Failed: 1,
+						Conditions: []batchv1.JobCondition{
+							{
+								LastProbeTime: metav1.Date(2024, 4, 29, 0o0, 0o0, 0o0, 0, time.UTC),
+								Type:          batchv1.JobFailed,
+								Reason:        "BackoffLimitExceeded",
+								Message:       "Job has reached the specified backoff limit",
+							},
+						},
+					},
+				}
+				return true, job, nil
+			},
+			mockListPods: func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				pod := &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("%s-%s-%s-1", project.Name, model.Name, modelVersion.ID),
+						Labels: map[string]string{
+							"job-name": fmt.Sprintf("%s-%s-%s", project.Name, model.Name, modelVersion.ID),
+						},
+					},
+					Status: v1.PodStatus{
+						Phase: v1.PodFailed,
+						ContainerStatuses: []v1.ContainerStatus{
+							{
+								Name: containerName,
+								State: v1.ContainerState{
+									Terminated: &v1.ContainerStateTerminated{
+										ExitCode: 1,
+										Reason:   "Error",
+										Message:  "CondaEnvException: Pip failed",
+									},
+								},
+							},
+						},
+					},
+				}
+				return true, &v1.PodList{
+					Items: []v1.Pod{*pod},
+				}, nil
+			},
+			want: models.ImageBuildingJobStatus{
+				State: models.ImageBuildingJobStateFailed,
+				Message: `Error
+
+Job conditions:
+┌───────────────────────────────┬────────┬──────────────────────┬─────────────────────────────────────────────┐
```

Comment:
i dont know if we really want to validate the table 😅  can we just check that `Job has reached the specified backoff limit` and `CondaEnvException: Pip failed` exist for example

Reply:
We can, but we will need to modify the assertion function to use some string comparison function. I'll keep it as is for now.